runtime/rasdump/javadump.cpp

```diff
@@ -2806,396 +2806,788 @@ JavaCoreDumpWriter::writeSharedClassSection(void)
 
-		_OutputStream.writeCharacters(
-			"\n2SCLTEXTBDA        Byte data bytes                           = "
-		);
-		_OutputStream.writeInteger(javacoreData.indexedDataBytes, "%zu");
+			_OutputStream.writeCharacters(
+					"\n2SCLTEXTPST        Percent Stale classes                     = "
+			);
+			_OutputStream.writeInteger(javacoreData.percStale, "%zu");
+			_OutputStream.writeCharacters("%\n");
+		
+			_OutputStream.writeCharacters(
+					"NULL\n"
+					"2SCLTEXTCPF        Cache is "
+			);
+			_OutputStream.writeInteger(javacoreData.percFull, "%zu");
+		
+			if (javacoreData.softMaxBytes == javacoreData.cacheSize) {
+				_OutputStream.writeCharacters("% full\n");
+			} else {
+				_OutputStream.writeCharacters("% soft full\n");
+			}
 
-		_OutputStream.writeCharacters(
-			"\n2SCLTEXTMDA        Metadata bytes                            = "
-		);
-		_OutputStream.writeInteger(javacoreData.otherBytes, "%zu");
+			_OutputStream.writeCharacters(
+					"NULL\n"
+					"1SCLTEXTCMST   Cache Memory Status\n"
+					"NULL           ------------------\n"
+					"1SCLTEXTCNTD       Cache Name                    Feature                  Memory type              Cache path\n"
+					"NULL\n"
+			);
+			_OutputStream.writeCharacters(
+					"2SCLTEXTCMDT       "
+			);
+			_OutputStream.writeCharacters(javacoreData.cacheName);
+			for (UDATA i = strlen(javacoreData.cacheName); i < 30; i++) {
+				_OutputStream.writeCharacters(" ");
+			}
+			if (J9_ARE_ALL_BITS_SET(javacoreData.feature, J9SH_FEATURE_COMPRESSED_POINTERS)) {
+				_OutputStream.writeCharacters("CR                       ");
+			} else if (J9_ARE_ALL_BITS_SET(javacoreData.feature, J9SH_FEATURE_NON_COMPRESSED_POINTERS)) {
+				_OutputStream.writeCharacters("Non-CR                   ");
+			} else {
+				_OutputStream.writeCharacters("Default                  ");
+			}
+			if (javacoreData.shmid == -2) {
+				_OutputStream.writeCharacters("Memory mapped file       ");
+			} else {
+				writeSharedClassIPCInfo("IPC Memory (id ", ")", javacoreData.shmid, 25);
+			}
+			_OutputStream.writeCharacters(javacoreData.cacheDir);
+			_OutputStream.writeCharacters("\n");
 
-		if (NO_CORRUPTION != javacoreData.corruptionCode) {
 			_OutputStream.writeCharacters(
-				"\n2SCLTEXTCOC        Corruption Code                           = "
+					"NULL\n"
+					"1SCLTEXTCMST   Cache Lock Status\n"
+					"NULL           ------------------\n"
+					"1SCLTEXTCNTD       Lock Name                     Lock type                TID owning lock\n"
+					"NULL\n"
 			);
-			_OutputStream.writeInteger(javacoreData.corruptionCode, "%zd");
 
+			writeSharedClassLockInfo(
+					"2SCLTEXTCWRL       Cache write lock              ", javacoreData.semid, javacoreData.writeLockTID
+			);
+		
+			writeSharedClassLockInfo(
+					"2SCLTEXTCRWL       Cache read/write lock         ", javacoreData.semid, javacoreData.readWriteLockTID
+			);
+		} else {
 			_OutputStream.writeCharacters(
```

Comment:
How about putting some of this into helper functions so it's not all duplicated between the two different types of output. Similarly for SH_CacheMap::printCacheStats().

Reply:
Added 3 helper functions